### ISSUE TITLE
pg middleware error -  transaction: exec: mismatched param and argument count - when trying to Update 

### ISSUE THREAD

**Issue by aleblanc70** [iris-contrib/middleware#383]
New to Iris and really like it!

Following your new article:
https://medium.com/@kataras/how-to-use-iris-and-postgresql-for-web-development-e8c46d72f1e6

I was able to use the pg middleware for testing. It seem I cannot do update! 
The error I got: **transaction: exec: mismatched param and argument count**

```
type User struct {
	Id         string    `json:"id" pg:"type=uuid,primary"`
	CreatedAt  time.Time `json:"createdAt" pg:"type=timestamp,default=clock_timestamp()"`
	UpdatedAt  time.Time `json:"updatedAt" pg:"type=timestamp,default=clock_timestamp()"`
	Email      string    `json:"email" pg:"type=varchar(255),username,unique"`
	Password   string    `json:"password" pg:"type=varchar(32),password"`
	FirstName  string    `json:"firstName" pg:"type=varchar(255)"`
	MiddleName *string   `json:"middleName" pg:"type=varchar(255),null"`
	LastName   string    `json:"lastName" pg:"type=varchar(255)"`
	Sex        *string   `json:"sex" pg:"type=char(1),null"`
}
```
```
func (c UserController) Update(ctx iris.Context) {
	var u User
	var errPg error
	err := ctx.ReadJSON(&u)
	if err != nil {
		ctx.StopWithProblem(iris.StatusBadRequest, iris.NewProblem().
			Title("User Update failure").DetailErr(err))
		return
	}

	userRepo := pg.Repository[User](ctx)
	_, errPg = userRepo.Update(ctx, u)
	if errPg != nil {
		if pg.IsErrNoRows(errPg) {
			ctx.StopWithStatus(iris.StatusNotFound)
		} else {
			ctx.StopWithError(iris.StatusInternalServerError, errPg)
		}
		return
	}
	ctx.StatusCode(iris.StatusOK)
}

```
From Postman I Put:
    {
        "id": "6c051e51-9c64-49f7-a6ce-977cd7b2030a",
        "email": "IronMan@gmail.com",
        "password": "Ironman",
        "firstName": "Tony",
        "middleName": "",
        "lastName": "Stark",
        "sex": "M"
    },

Same error if I add my CreatedAt and UpdatedAt in my json 

I tried to debug further, and it remove my audit dates because there are timestamp and put my id at the end not sure what it try to do.

Any help will be appreciated!

My project is here:
https://github.com/aleblanc70/events/blob/main/web/main.go

Thanks!

**Comment by aleblanc70** [comment]
I tried with my event model

```
{
    "id": "a613ab5d-9cc1-45ca-ba1d-9bc8e4fcff2f",
    "title": "Ironman 2024",
    "description": "IronMan"
}
```


When I reach the code: 
```
func (db *DB) updateTableRecord(ctx context.Context, value any, columnsToUpdate []string, primaryKey *desc.Column) (int64, error) {
	// build the SQL query and arguments using the table definition and its primary key.
	query, args, err := desc.BuildUpdateQuery(value, columnsToUpdate, primaryKey)
	if err != nil {
		return 0, err
	}

	// execute the query using db.Exec and pass in the primary key values as a parameter
	tag, err := db.Exec(ctx, query, args...)
	if err != nil {
		return 0, err
	}

	return tag.RowsAffected(), nil
}

```
My **query** have 4 arguments and my **args** got 3 element only. Look like the id is causing the issue maybe... 

UPDATE "events" SET title = $1,description = $2,id = $3 WHERE "id" = $4;

Should I have Id twice in my args parameter ?

**Comment by aleblanc70** [comment]
Fix in pg pkg 1.0.1 thanks!
